File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOBpCalibrationParamModel.swift
//
//  IDOBpCalibrationParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Blood pressure calibration event number
@objcMembers
public class IDOBpCalibrationParamModel: NSObject, IDOBaseModel {
    /// 1: Blood pressure calibration settings
    /// 2: Blood pressure calibration query result
    public var flag: Int
    /// Systolic pressure
    public var diastolic: Int
    /// Diastolic pressure
    public var systolic: Int
    
    enum CodingKeys: String, CodingKey {
        case flag = "flag"
        case diastolic = "diastolic"
        case systolic = "systolic"
    }
    
    public init(flag: Int,diastolic: Int,systolic: Int) {
        self.flag = flag
        self.diastolic = diastolic
        self.systolic = systolic
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOBpCalibrationParamModel.kt
//
//  IDOBpCalibrationParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Blood pressure calibration event number

open class IDOBpCalibrationParamModel(flag: Int, diastolic: Int, systolic: Int) : IDOBaseModel {

    /// 1: Blood pressure calibration settings
    /// 2: Blood pressure calibration query result
    @SerializedName("flag")
    var flag: Int = flag

    /// Systolic pressure
    @SerializedName("diastolic")
    var diastolic: Int = diastolic

    /// Diastolic pressure
    @SerializedName("systolic")
    var systolic: Int = systolic


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    